### PR TITLE
provide masks for a subselection of a shapefile's polygon

### PR DESCRIPTION
**Pull request by BenMGeo**
Just few changes to @valeriupredoi's inclusion of multiple polygons from a shapefile.
This way a user can provide indices instead of a "last" polygon.

- [ ] To Do: decide on sorting or not sorting the polygons (I would not!) and changing the index for standard masks accordingly.

### CODE REVIEW

**Comment by BenMGeo** [comment]
Closed as included directly in #389 389

File: esmvalcore/preprocessor/_mask.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def mask_landsea(cube, fx_files, mask_out, always_use_ne_mask=False):
             logger.debug("Applying land-sea mask: sftof")
         else:
             if cube.coord('longitude').points.ndim < 2:
-                cube = _mask_with_shp(cube, shapefiles[mask_out], 1)
+                cube = _mask_with_shp(cube, shapefiles[mask_out], [0])  # TODO: this index might be wrong without sorting!!!
```
[the remark this answers is not in the document]
Comment (BenMGeo) [reply esmvalcore/preprocessor/_mask.py]: I think you would not need the `,`. But anyways.... Good you found the highlights ;)

[the remark this answers is not in the document]
Comment (valeriupredoi) [reply esmvalcore/preprocessor/_mask.py]: pythonic man :grin:

**Comment by valeriupredoi** [comment]
cheers @BenMGeo :beer:

**Pull request closed without merging**